app/controllers/report_controller/reports/editor.rb

```diff
@@ -9,7 +9,7 @@ module ReportController::Reports::Editor
   DEFAULT_PDF_PAGE_SIZE = "US-Letter".freeze
 
   MAX_REPORT_COLUMNS = 100 # Default maximum number of columns in a report
-  GRAPH_MAX_COUNT = 10
+  GRAPH_MAX_COUNT = ::Settings.reporting.chart_top_values || 10
```

Comment:
This is happening at constant time, so if the user changes it you'd have to reboot the worker/server.  Instead of a constant, this should probably move to a method to allow the runtime lookup.

Additionally, we try to avoid having "hidden" values in the settings.  Instead, it's probably preferable to _only_ use settings, and then just set the value in the settings to 10.

Reply:
@Fryguy I'm afraid that I'm not versed to come up with the best way to do what you are suggesting, but PTAL at my latest force-push. It appears to achieve the dynamic behavior on UI.

In the meantime, I opened this [PR on Core](https://github.com/ManageIQ/manageiq/pull/22632) to support this logic.